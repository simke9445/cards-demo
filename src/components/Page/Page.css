.half-circle {
  width: 50px;
  height: 100px;
  border-bottom-right-radius: 100px;
  border-top-right-radius: 100px;
  background-color: lightgray;
  transition: width 1s linear;
  transform: translate(-25px);
}

.half-circle:hover {
  background-color: gray;
  transform: translate(0px);
}

.half-circle-rotated {
  width: 50px;
  height: 100px;
  border-bottom-right-radius: 100px;
  border-top-right-radius: 100px;
  background-color: lightgray;
  transition: width 1s linear;
  transform: translate(25px) rotate(180deg);
}

.half-circle-rotated:hover {
  background-color: gray;
  transform: translate(0px) rotate(180deg);
}