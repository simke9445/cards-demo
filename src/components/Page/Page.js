import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { Icon } from '@blueprintjs/core';

import "./Page.css";

const styles = {
  topBar: {
    height: '7.5%',
    width: '90%',
    backgroundColor: 'white',
    boxShadow: 'rgba(0, 0, 0, 0.15) 0px 3px 14px',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  container: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#F5F5F5',
  },
  empty: {
    width: '20%',
    visibility: 'hidden',
  },
  topBarHeader: {
    width: '20%',
    paddingLeft: '14px',
    textAlign: 'left',
    color: '#B2B6B9',
  },
  progressBarContainer: {
    width: '60%',
  },
  progressBarMeter: {
    display: 'flex',
    backgroundColor: '#A06FEF',
  },
  contentContainer: {
    flex: '1',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%',
  },
  halfCircle: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    cursor: 'pointer',
  },
  leftArrow: {
    fontSize: '3em',
    color: 'white',
    transform: 'translate(-10px)',
  },
  rightArrow: {
    fontSize: '3em',
    color: 'white',
    transform: 'translate(-10px)',
  }
}

const Page = ({ children, hasNext, hasPrevious, pageIndex, onNext, onPrevious }) => 
  <div style={styles.container}>
    <div className="pt-callout" style={styles.topBar}>
      <h3 style={styles.topBarHeader}><FormattedMessage id={'page.title'}/></h3>
      <div style={styles.progressBarContainer}>
        <div className="pt-progress-bar pt-no-stripes">
          <div className="pt-progress-meter" style={{ ...styles.progressBarMeter, width: `${100*pageIndex/4}%` }}></div>
        </div>
      </div>
      <div style={styles.empty}>{null}</div>
    </div>
    <div style={styles.contentContainer}>
      <div className="half-circle" style={styles.halfCircle} onClick={hasPrevious ? onPrevious : null}>
        <Icon iconName="chevron-left" style={styles.leftArrow}/>
      </div>
        {children}
      <div className="half-circle-rotated" style={styles.halfCircle} onClick={hasNext ? onNext : null}>
        <Icon iconName="chevron-left" style={styles.rightArrow}/>
      </div>
    </div> 
  </div>

Page.propTypes = {
  pageIndex: PropTypes.number.isRequired,
  onNext: PropTypes.func,
  onPrevious: PropTypes.func,
  hasNext: PropTypes.bool,
  hasPrevious: PropTypes.bool,
};

export default Page;