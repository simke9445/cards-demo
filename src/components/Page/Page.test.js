import React from 'react';
import Page from './Page';
import { shallow } from 'enzyme';

it('Page renders without crashing', () => {
  const pageIndex = 1;
  const onNext = jest.fn();
  const onPrevious = jest.fn();

  const wrapper = shallow(
    <Page 
      pageIndex={pageIndex}
      hasNext={true}
      hasPrevious={true}
      onNext={onNext}
      onPrevious={onPrevious}
    ></Page>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('div.half-circle').length).toEqual(1);
  expect(wrapper.find('div.half-circle-rotated').length).toEqual(1);
  wrapper.find('div.half-circle').simulate('click');
  expect(onPrevious).toBeCalled();
  wrapper.find('div.half-circle-rotated').simulate('click');
  expect(onNext).toBeCalled();
});
