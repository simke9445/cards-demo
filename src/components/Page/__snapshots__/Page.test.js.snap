// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders without crashing 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <div
    className="pt-callout"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "boxShadow": "rgba(0, 0, 0, 0.15) 0px 3px 14px",
        "display": "flex",
        "height": "7.5%",
        "justifyContent": "space-between",
        "width": "90%",
      }
    }
  >
    <h3
      style={
        Object {
          "color": "#B2B6B9",
          "paddingLeft": "14px",
          "textAlign": "left",
          "width": "20%",
        }
      }
    >
      <FormattedMessage
        id="page.title"
        values={Object {}}
      />
    </h3>
    <div
      style={
        Object {
          "width": "60%",
        }
      }
    >
      <div
        className="pt-progress-bar pt-no-stripes"
      >
        <div
          className="pt-progress-meter"
          style={
            Object {
              "backgroundColor": "#A06FEF",
              "display": "flex",
              "width": "25%",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "visibility": "hidden",
          "width": "20%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": "1",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <div
      className="half-circle"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "cursor": "pointer",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <Blueprint.Icon
        iconName="chevron-left"
        style={
          Object {
            "color": "white",
            "fontSize": "3em",
            "transform": "translate(-10px)",
          }
        }
      />
    </div>
    <div
      className="half-circle-rotated"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "cursor": "pointer",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <Blueprint.Icon
        iconName="chevron-left"
        style={
          Object {
            "color": "white",
            "fontSize": "3em",
            "transform": "translate(-10px)",
          }
        }
      />
    </div>
  </div>
</div>
`;
